### PR TITLE
[modbus] fixed karaf build

### PR DESCRIPTION
**Pull request by mrbig**
The modbus binding is dependant on the io.transport.modbus bundle, wich is
also part of the openhab2-addons. By default these bundles are not in
the build repositories.

Previously this has been fixed by a direct file link, but that seems to
be unstable especially under travis builds.

Here I propose to use the direct maven repository of the required bundle.

@ssalonen please let me know what do you think about this change?

### CODE REVIEW

**Comment by ssalonen** [comment]
@mrbig to be honest I am probably not the right person the review this.

I am not proficient enough with the karaf features... Is there any other bundle that has similar setup?

File: bundles/org.openhab.binding.modbus/src/main/feature/feature.xml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <features name="org.openhab.binding.modbus-${project.version}" xmlns="http://karaf.apache.org/xmlns/features/v1.4.0">
-    <repository>file:${basedirRoot}/bundles/org.openhab.io.transport.modbus/target/feature/feature.xml</repository>
+    <repository>mvn:org.openhab.core.features.karaf/org.openhab.core.features.karaf.openhab-core/${project.version}/xml/features</repository>
```
Comment (J-N-K) [inline bundles/org.openhab.binding.modbus/src/main/feature/feature.xml]: Why is this needed at all?

File: bundles/org.openhab.binding.modbus/src/main/feature/feature.xml [hunk 2]
```diff
@@ -1,6 +1,7 @@
...
+    <repository>mvn:org.openhab.addons.bundles/org.openhab.io.transport.modbus/${project.version}/xml/features</repository>
```
Comment (J-N-K) [inline bundles/org.openhab.binding.modbus/src/main/feature/feature.xml]: The problem with this approach is: If the transport bundle has not been compiled yet (which is the only reason I can image why the `file`-repo fails), an old version is used. That could be wrong.

File: bundles/org.openhab.binding.modbus/src/main/feature/feature.xml (hunk 1 above)
Comment (mrbig) [reply bundles/org.openhab.binding.modbus/src/main/feature/feature.xml]: You are right the core features repo can be omitted safely.

**Comment by wborn** [comment]
A more generic approach to solve these incremental build issues for all add-ons might be to let Maven also build all projects the changed projects depend on using the `-am` switch :

```
 -am,--also-make                        If project list is specified, also
                                        build projects required by the
                                        list
```

**Comment by mrbig** [comment]
Thanks @wborn I'm also thinking that this might be a travis build issue. Currently buildci.sh changes to the module directory, and tries to build the single addon: https://github.com/openhab/openhab2-addons/blob/aa190bec2e9e142a335d12ef50e71a95aa7d395a/buildci.sh#L51

Could this be changed to run the build with -am?

**Comment by wborn** [comment]
> Could this be changed to run the build with -am -pl?

Yes `buildci.sh` is only used by Travis so it makes to sense to use these switches here so Maven can use the dependencies of Maven projects to figure out what needs to be build.

I think a similar approach is used by Jenkins with incremental Maven build jobs.

**Comment by mrbig** [comment]
@wborn I've submitted PR #6337 with an example how buildci.sh could be updated to build the single bundle but with dependencies. This change could render this PR obsolete.

**Comment by wborn** [comment]
Obsolete because of #6337
